README.md

```diff
@@ -1,3 +1,149 @@
+
+```bash
+|install ~pocwet %example-desk
+|mount %example-desk
+```
+
+You can then view the files associated with this example on your machine at the
+file path `/path/to/ship/example-desk`. Please note that this method **will only
+install Hoon files**; to view and experiment with React front-end files, please
+read on.
+
+## Build ##
+
+### First-time Setup ###
+
+To build one of these example from scratch (which is essential for React
+development), follow these instructions:
+
+1. Download and extract the repo source code [from GitHub][gitsrc], e.g.
+   at `/path/to/repo-source/`.
+2. Follow the instructions for [creating a fake ship][fakeship], e.g.
+   hosting it at `/path/to/fake-ship/`.
+3. When booting up your fake ship, take note of the following line in
+   the introductory digest:
+   ```
+   http: web interface live on http://localhost:XYZ
+   ```
+   The `XYZ` string on this line is port number your ship is using for its HTTP
+   interface. Save this string for use in the subsequent steps.
+4. In your fake ship's `dojo`, enter the following commands:
+   ```
+   |new-desk %example-desk
+   |mount %example-desk
+   ```
+5. In a separate terminal session, enter the following commands:
+   ```bash
+   rm -rI /path/to/fake-ship/example-desk/*
+   cp -RL /path/to/repo-source/example-directory/full-desk/* /path/to/fake-ship/example-desk/
+   ```
+6. Back in your fake ship's `dojo`, enter the following commands:
+   ```
+   |commit %example-desk
+   |install our %example-desk
+   ```
+7. Again in your fake ship's `dojo`, run the following command and copy the
+   result:
+   ```
+   +code
+   ```
+   This is your fake ship's password, which will be used for authentication
+   from your web browser.
+8. Open a web browser and navigate to the HTTP address discussed in step 3.
+9. Paste the ship's password from step 7 into the form that appears and hit
+   enter.
+0. If the installation was successful, you should see a tile for the example
+   on the home page that appears.
+
+If the example has a React interface, further follow these instructions to
+set up a locally-hosted front-end interface:
+
+11. In a terminal session, enter the following commands (where `XYZ` is from
+    step 3):
+    ```bash
+    cd /path/to/repo-source/example-directory/ui/
+    npm install
+    echo "VITE_SHIP_URL=http://127.0.0.1:XYZ" >> .env.local
+    npm run dev
+    ```
+    The final command will generate a text digest containing a line of the form:
+    ```
+    > Local: http://localhost:ABC/apps/example-desk/
+    ```
+    The web address listed here is hosting location for your local React
+    interface. Save this string for use in the subsequent steps.
+12. Open a web browser and navigate to the web address from the step 11.
+    If the local hosting was successful, you should see the interface for the
+    example in your browser.
+13. If your browser encounters an error during the previous step, then you may
```

Comment:
Can you expand on the error?

Reply:
I've actually never encountered this error with CORS; I just copied this section forward from the [`chat-app` UI readme](https://github.com/urbit/docs-examples/blame/main/chat-app/react-frontend/README.md#L5).

@matildepark: My `git blame` suggests that you first introduced this workaround for CORS failures. Would you happen to know if the `|cors-approve` step is still necessary for modern `create-landscape-app` Vite-proxied apps? If so, do you remember what sort of error crops up in the browser if it isn't set on a ship that needs it?

Reply:
@matildepark is right you would only ever need this if connecting to a ship from a completely different origin, which currently is not a very common pattern. I think this can probably be relegated to another location like https://github.com/urbit/js-http-api.

Reply:
Isn’t localhost considered a different origin?

Reply:
Ohhh so you mean developing on a dev ship hosted locally vs. proxying onto a running ship. I understand. 

On the other hand, I would expect Scene’s API connection to need CORS and I don’t think it does.

Reply:
Maybe we just fixed that client side actually. It was a while ago.

Reply:
Thanks @matildepark and @arthyn for clarifying all of this. It sounds like this isn't an issue for the most common developer workflow (i.e. both Urbit ship and React FE on `localhost`), so I'll remove this step from the guide.

Reply:
I have tried building, globbing and uploading hut and journal FEs: hut seems to work fine but journal produces a "minified react error #130" error in both chromium and firefox and is just a blank page. I'm just cloning your repo then running `npi i` and `npm run build` - I don't think I'm doing anything wrong

Reply:
Thanks @tinnus-napbus; I'm glad you tested globbing because I just assumed that if it built it would work! I'll look into this and get it fixed.

Reply:
This issue should be fixed with the latest commit; I tested both the Vite and globbed versions in Firefox. Let me know if you see any other issues!